### PR TITLE
feat: v4 rules card, "TOKEN" section [1/n]

### PR DESCRIPTION
**Pull request by johnnyd-eth**
Screenshot 2024-07-15 at 10 57 12 pm

Screenshot 2024-07-15 at 10 57 36 pm

### CODE REVIEW

File: src/utils/tokenSymbolText.ts [hunk 1]
```diff
@@ -12,13 +12,13 @@ export const tokenSymbolText = ({
   plural?: boolean
   includeTokenWord?: boolean
 }) => {
-  const tokenTextSingular = capitalize ? t`Token` : t`token`
-  const tokenTextPlural = capitalize ? t`Tokens` : t`tokens`
-  const tokenText = plural ? tokenTextPlural : tokenTextSingular
+  const defaultTokenTextSingular = capitalize ? t`Token` : t`token`
+  const defaultTokenTextPlural = capitalize ? t`Tokens` : t`tokens`
```
Comment (johnnyd-eth) [inline src/utils/tokenSymbolText.ts:16]: Just made the naming here more clear, no functional changes

File: src/packages/v2v3/components/V2V3Project/ProjectDashboard/components/CyclesPayoutsPanel/hooks/useConfigurationPanel/useFormatConfigurationExtensionSection.ts [hunk 2]
```diff
@@ -1,14 +1,14 @@
 import { t } from '@lingui/macro'
+import { flagPairToDatum } from 'components/Project/ProjectHeader/utils/flagPairToDatum'
```
Comment (aeolianeth) [inline src/packages/v2v3/components/V2V3Project/ProjectDashboard/components/CyclesPayoutsPanel/hooks/useConfigurationPanel/useFormatConfigurationExtensionSection.ts:2]: Is ProjectHeader the best place for this? 🤔

File: src/packages/v4/models/rulesetMetadata.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { RedemptionRate, ReservedRate } from "juice-sdk-core";
+
+export type RulesetMetadata = {
```
Comment (aeolianeth) [inline src/packages/v4/models/rulesetMetadata.ts:3]: SDK has `JBRulesetMetadata`

File: src/packages/v4/views/V4ProjectDashboard/V4ProjectTabs/CyclesPayoutsPanel/hooks/useV4FormatConfigurationTokenSection.ts [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+      tokenSymbolText({
+        tokenSymbol: tokenSymbolRaw,
+        capitalize: false,
+        plural: true,
+      }),
+    [tokenSymbolRaw],
+  )
+  const decayRateFloat = ruleset?.decayRate.toFloat()
+  const currentTotalIssuanceRate = ruleset?.weight.toFloat()
+  const queuedTotalIssuanceRate = queuedRuleset ?
```
Comment (aeolianeth) [inline src/packages/v4/views/V4ProjectDashboard/V4ProjectTabs/CyclesPayoutsPanel/hooks/useV4FormatConfigurationTokenSection.ts:39]: This hook feels weird to me. Why is it combining the queued ruleset and current ruleset. Could we just have a hook that takes a ruleset, does the formatting, and we just pass in the current or queued depending on the selection?

File: src/packages/v2v3/components/V2V3Project/ProjectDashboard/components/CyclesPayoutsPanel/hooks/useConfigurationPanel/useFormatConfigurationExtensionSection.ts (hunk 2 above)
Comment (johnnyd-eth) [reply src/packages/v2v3/components/V2V3Project/ProjectDashboard/components/CyclesPayoutsPanel/hooks/useConfigurationPanel/useFormatConfigurationExtensionSection.ts:2]: That's my mistake, was meant to put it in ProjectTabs, moved it there now.

**Pull request merged**